app/controllers/api/users.php

```diff
@@ -806,6 +818,15 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
 
         $newPassword = Auth::passwordHash($password, Auth::DEFAULT_ALGO, Auth::DEFAULT_ALGO_OPTIONS);
 
+        $passwordDictionary = $project->getAttribute('auths', [])['passwordDictionary'] ?? false;
```

Comment:
Can we abstract the logic behind the framework validator? We can pass both the project document and dictionary into the validator is DI and using a callback.

Reply:
we can, but not in the parameter validation because, we will need the project 's `auths/passwordDictionary` value

Reply:
we will have to do it like the password history validator

Reply:
updated